### ISSUE TITLE
errata_tool_user: add type parameter

### ISSUE THREAD

**Issue by ktdreyer** [ktdreyer/errata-tool-ansible#59]
The REST API supports a `type` parameter for Errata Tool user accounts. The options are `Person`, `KerberosPrincipal`, or `MailingList`.

We should expose this parameter in the `errata_tool_user` module.

If the user does not specify a `type`, the Errata Tool server guesses what user type to use according to some conditions. The Errata Tool developers have documented that this guessing behavior is temporary and the developers expect all clients to explicitly specify a `type`.

If we make the `type` parameter mandatory, **we will break existing user playbooks that call `errata_tool_user`**, but maybe this is ok for now. The guessing behavior breaks in dev environments that are not staging or prod because it relies on finger.

**Comment by ktdreyer** [comment]
@pbortlov and @Michkov FYI, I'm thinking we should take this breaking change now in `errata_tool_user` (making "type" mandatory)

**Comment by Michkov** [comment]
That's fine for us. How about set `Person` as default? In errata web ui it's selected by default.

**Comment by pbortlov** [comment]
@ktdreyer @Michkov I don't think we have to add `Person` as default. Errata is clever to solve it for us.
Errata Docs:
NOTE: If the type parameter is not specified, the system will attempt to guess the correct type of user (Note that this is intended to be temporary until the user generation tools start using type): If the user has a / in the login_name, it is assumed to be KerberosPrincipal If not, and the user has a - in the login_name, it is assumed to be MailingList Otherwise, it will be created as a Person, aka regular Errata Tool user.

**Comment by ktdreyer** [comment]
I like the idea of defaulting to "Person" since that matches the web UI.

@pbortlov unfortunately that Errata Tool documentation does not match the server's implementation, and I filed RHELWF-1135 to track that problem server-side. In my testing, I always get `MailingList` users if I do not specify a `type`.

**Comment by ktdreyer** [comment]
Here is the behavior I'm proposing:

A) The playbook author specifies `type: null` (or just omits the `type` parameter for `errata_tool_user` entirely): If the ET user account does not yet exist, send a POST request with "type: Person" (in the spirit of RHELWF-1135). If we're editing an existing account, do not alter the `type` at all.

B) The playbook author specifies `type`: `Person`: If the user account does not yet exist, send a POST request with "type: Person". If we're editing an existing account, ensure that `type` is a `Person` and edit it with a PUT request if it is not what the user specifies.
